projects/addon-editor/extensions/image-editor/image-editor.extension.ts

```diff
@@ -78,5 +81,57 @@ export const createImageEditorExtension = (injector: Injector): Node => {
                         }),
             };
         },
+
+        addProseMirrorPlugins() {
+            return [
+                new Plugin({
+                    props: {
+                        handleDOMEvents: {
+                            paste: pasteImage,
+                            drop: pasteImage,
+                        },
+                    },
+                }),
+            ];
+        },
     });
 };
+
+function pasteImage(view: EditorView, event: ClipboardEvent | DragEvent): void {
+    const dataTransfer =
+        event instanceof DragEvent ? event.dataTransfer : event.clipboardData;
+    const hasFiles = dataTransfer?.files?.length ?? 0;
+
+    if (!hasFiles) {
```

Comment:
I think this check is unnecessary, we test for `images.length` later. It only saves us from unnecessary `Array.from` and `filter` which is negligible.

Reply:
fixed